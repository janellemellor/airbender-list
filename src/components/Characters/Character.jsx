import React from 'react';
import PropTypes from 'prop-types';

const Character = ({ image, name, affiliation }) => (
  <>
    <img src={image} alt={name} />
    <h3>{name}</h3>
    <p>{affiliation}</p>
  </>
);

Character.propTypes = {
  image: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  affiliation: PropTypes.string.isRequired
};

export default Character;
