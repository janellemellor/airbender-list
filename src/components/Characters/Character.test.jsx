import React from 'react';
import { shallow } from 'enzyme';
import Character from './Character.jsx';

describe('Character', () => {
  it('matches the Character snapshot', () => {
    const wrapper = shallow(<Character image='https://vignette.wikia.nocookie.net/avatar/images/5/51/46th_Earth_King.png/revision/latest?cb=20130627160441' name='46th Earth King' affiliation='Earth Kingdom Earth Kingdom Royal Family' />);
    expect(wrapper).toMatchSnapshot();
  }); 
});

