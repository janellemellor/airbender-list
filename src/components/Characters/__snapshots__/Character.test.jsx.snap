// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character matches the Character snapshot 1`] = `
<Fragment>
  <img
    alt="46th Earth King"
    src="https://vignette.wikia.nocookie.net/avatar/images/5/51/46th_Earth_King.png/revision/latest?cb=20130627160441"
  />
  <h3>
    46th Earth King
  </h3>
  <p>
    Earth Kingdom Earth Kingdom Royal Family
  </p>
</Fragment>
`;
